etcdserver/security/security.go

```diff
@@ -239,16 +243,20 @@ func (s *Store) UpdateUser(user User) (User, error) {
 }
 
 func (s *Store) AllRoles() ([]string, error) {
+	nodes := []string{GuestRoleName, RootRoleName}
 	resp, err := s.requestResource("/roles/", false)
 	if err != nil {
+		if e, ok := err.(*etcderr.Error); ok {
+			if e.ErrorCode == etcderr.EcodeKeyNotFound {
+				return nodes, nil
```

Comment:
The doc says that guest role will not be created until security is enabled. https://github.com/coreos/etcd/blob/master/Documentation/security_api.md#roles